core/src/provider.rs

```diff
@@ -34,6 +38,26 @@ pub struct WrappedLog {
     pub block_timestamp: Option<U256>,
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TraceCall {
+    pub from: Address,
+    pub gas: String,
+    #[serde(rename = "gasUsed")]
+    pub gas_used: U256,
+    pub to: Address,
+    pub input: Bytes,
+    pub value: U256,
+    #[serde(rename = "type")]
+    pub typ: String,
```

Comment:
Gotta love reserve words. Let's rename something like `call_type` rather than `typ`.


```suggestion
    pub call_type: String,
```

Reply:
I mimicked what ethers-rs already used which is `typ`, but this has been refactored now.